services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -1664,13 +1667,24 @@ Version addVersionAndTopicOnly(
         versionSwapDeferred);
     Version newVersion = result.getSecond();
     if (result.getFirst()) {
+      // fail early if canary region push is enabled but source region is not set.
+      if (canaryRegionPush && StringUtils.isEmpty(newVersion.getNativeReplicationSourceFabric())) {
```

Comment:
shouldnt we look up cluster level config if store is not set?

Reply:
No we don't need to. The `NativeReplicationSourceFabric` should have been decided and set by the `AddTopicAndVersionOnly` method above ([line](https://github.com/linkedin/venice/blob/faaea30a22ccc36756659f6a400d19987c91e574/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java#L1651)) using various values, e.g. cluster config and store config.